kit/cli/viper.go

```diff
@@ -60,20 +59,6 @@ func NewCommand(v *viper.Viper, p *Program) *cobra.Command {
 	// This normalizes "-" to an underscore in env names.
 	v.SetEnvKeyReplacer(strings.NewReplacer("-", "_"))
 
-	if configPath := v.GetString("CONFIG_PATH"); configPath != "" {
-		switch strings.ToLower(path.Ext(configPath)) {
-		case ".json", ".toml", ".yaml", ".yml":
-			v.SetConfigFile(configPath)
-		case "":
-			v.AddConfigPath(configPath)
-		}
```

Comment:
I think this should have just been changed to:

```
		switch strings.ToLower(path.Ext(configPath)) {
 		case ".json", ".toml", ".yaml", ".yml":
 			v.SetConfigFile(configPath) // maybe also check if configPath is actually a directory, and if so, error message.
 		default:
 			v.AddConfigPath(configPath)
 		}
```
if someone has '*.yaml' as their config directory they deserve problems.

Reply:
I did consider that change originally. The question to answer is: what do we want the behavior to be if a user sets `INFLUXD_CONFIG_PATH` to point at a file that doesn't end in one of our supported extensions?

As it is now, the PR would detect that the env is pointing directly at a file, pass that file along to `viper`, and then `viper` would raise an error about the unsupported extension.

With the suggested change, the code would detect the extension mismatch and add the path to `viper`'s search path. `viper` would then stat `${INFLUXD_CONFIG_PATH}/config.${ext}` for each of its supported extensions (all failing). Another error-handling block below the changes here would ensure the config-loading failure didn't crash the program, so `influxd` would boot up with default config values.

Reply:
If the user specifies a config file that doesn't have the right extension, I want `influxd` to refuse to boot.

Reply:
If they specify a directory (or a path that doesn't point to a file or directory) I'm fine with them getting default config.

Reply:
Implemented below